### PR TITLE
DOP-4036: Add more term mappings

### PR DESCRIPTION
**Pull request by casthewiz**
DOP-4036

Updates the term mappings, making some edits to `nodejs` entries. Edits have been cleared with @sarahemlin - original entries may oversample different docsets.

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -8,6 +8,7 @@ trigger:
     include:
       - 'refs/tags/v*'
       - 'refs/heads/main'
+      - 'refs/heads/DOP-4036'
```
Comment (casthewiz) [inline .drone.yml:11]:
```suggestion
```

**Review by schmalliso — APPROVED** [review]
:shipit: 🚂 🇫🇷

File: src/data/term-result-mappings.ts [hunk 2]
```diff
@@ -87,6 +87,97 @@ const resultMapping: ResultMapping = {
...
+  ],
+  inc: [
+    'reference/operator/update/inc',
+    'reference/method/db.collection.updateMany',
+    'reference/method/db.collection.findAndModify',
+  ],
+  $pull: ['reference/operator/update/pull', 'reference/operator/update/pullAll', 'reference/operator/update'],
+  $reduce: ['reference/operator/aggregation/reduce', 'reference/operator/aggregation'],
+  $all: ['reference/operator/query/all', 'reference/operator/query/elemMatch', 'reference/operator/query/and'],
+  $facet: ['reference/operator/aggregation/facet', 'reference/operator/aggregation/bucket', 'atlas/atlas-search/facet'],
```
Comment (seungpark) [inline src/data/term-result-mappings.ts]: good catch on no string in txt! :D

**Review by seungpark — APPROVED** [review]
LGTM

File: src/data/term-result-mappings.ts [hunk 3]
```diff
@@ -87,6 +87,97 @@ const resultMapping: ResultMapping = {
...
+  cond: ['reference/operator/aggregation/cond', 'tutorial/query-documents', 'reference/operator/query/expr'],
+  node: ['quick-start/download-and-install', 'fundamentals/promises'],
+  'node.js driver': ['quick-start/download-and-install', 'fundamentals/promises'],
+  nodejs: ['quick-start/download-and-install', 'fundamentals/promises'],
+  $filter: [
+    'reference/operator/aggregation/filter',
+    'reference/operator/aggregation',
+    'reference/operator/projection/positional',
+  ],
+  $match: ['reference/operator/aggregation/match', 'core/aggregation/', 'core/aggregation-pipeline-optimization/'],
```
Comment (schmalliso) [inline src/data/term-result-mappings.ts:99]: Just discovered that core/aggregation isn't a page anymore - it's at /aggregation

**Pull request merged**